### ISSUE TITLE
Windows Taildrop should use SHGetKnownFolderPath for destination folder

### ISSUE THREAD

**Issue by DentonGentry** [tailscale/tailscale#2112]
Originally reported https://forum.tailscale.com/t/bug-in-taildrop-to-windows/726

There is a bug in the taildrop windows client when the windows machine has onedrive installed. In this instance, the desktop folder is not in \users<name>\Desktop but instead it’s in \users<name>\Onedrive\Desktop. Taildrop should be using the SHGetKnownFolderPath API with FOLDERID_Desktop see here for more details: [SHGetKnownFolderPath function (shlobj_core.h) - Win32 apps | Microsoft Docs](https://docs.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath)

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_38zwx)

**Comment by DentonGentry** [comment]
Found the place, I think: files.go

```
func copyFile(wf apitype.WaitingFile) error {
        desktop := filepath.Join(os.Getenv("UserProfile"), "Desktop")
```

**Comment by DentonGentry** [comment]
Appears to be straightforward:

```
$ git diff win/ui/files.go
diff --git a/win/ui/files.go b/win/ui/files.go
index 7d8dcb123..bf55f054e 100644
--- a/win/ui/files.go
+++ b/win/ui/files.go
@@ -12,6 +12,7 @@ import (
        "path/filepath"
        "strings"

+       "golang.org/x/sys/windows"
        "tailscale.com/client/tailscale"
        "tailscale.com/client/tailscale/apitype"
 )
@@ -49,7 +50,10 @@ func (a *app) fetchFiles() {
 }

 func copyFile(wf apitype.WaitingFile) error {
-       desktop := filepath.Join(os.Getenv("UserProfile"), "Desktop")
+       desktop, err := windows.KnownFolderPath(windows.FOLDERID_Desktop, 0)
+       if err != nil {
+               return err
+       }
        fi, err := os.Stat(desktop)
        if err != nil {
                return err
```

I was able to use tailscale to copy the installer file to the Windows host, install it, then use tailscale to copy another file to show that it still works.

I don't have OneDrive on any system to test it further.

**Comment by clarkezone** [comment]
How easy is it to build a private?  Happy to test

**Comment by DentonGentry** [comment]
This change is included in unstable build 1.9.164 at https://pkgs.tailscale.com/unstable/#windows

**Comment by clarkezone** [comment]
Just tested it and confirmed it fixes the issue.  Thx for rapid response!
